### ISSUE TITLE
Use rootless images where possible

### ISSUE THREAD

**Issue by czunker** [mondoohq/mondoo-operator#423]
**Is your feature request related to a problem? Please describe.**
Currently, all our images are running as root.

**Describe the solution you'd like**
Use the new rootless images where possible:
https://github.com/mondoohq/client/pull/151

Also set `runAsNonRoot: true` and `runAsUser`.

**Describe alternatives you've considered**
-

**Additional context**
The node scan seems to need root. Instead of root, perhaps some capabilities would be sufficient?
Verify this: https://github.com/mondoohq/client/pull/151#issuecomment-1165289510

**Comment by czunker** [comment]
No matter whether I run the node scan with root or not, I get the same score.
So, seems we are good to go and give it a try.

root node scan:
```
{
  "allowPrivilegeEscalation": false,
  "readOnlyRootFilesystem": true
}

```

```
Summary
========================

Target:     minikube
Score:      C    55/100     (100% completed)
✓ Passed:   ██████ 39% (41)
✕ Failed:   ████ 27% (29)
! Errors:   █████ 33% (35)
» Skipped:  1% (1)

Policies:
F   0  Platform Vulnerability Policy by Mondoo
.  ..  Platform Overview Information by Mondoo
B  65  Linux Security Baseline by Mondoo
A 100  Platform End-of-Life Policy by Mondoo

```

rootless node scan:
```
{
  "allowPrivilegeEscalation": false,
  "readOnlyRootFilesystem": true,
  "runAsNonRoot": true,
  "runAsUser": 101
}

```

```
Summary
========================

Target:     minikube
Score:      C    55/100     (100% completed)
✓ Passed:   ██████ 39% (41)
✕ Failed:   ████ 27% (29)
! Errors:   █████ 33% (35)
» Skipped:  1% (1)

Policies:
F   0  Platform Vulnerability Policy by Mondoo
.  ..  Platform Overview Information by Mondoo
B  65  Linux Security Baseline by Mondoo
A 100  Platform End-of-Life Policy by Mondoo

```

rootless cap drop node scan:
```
{
  "allowPrivilegeEscalation": false,
  "capabilities": {
    "drop": [
      "ALL"
    ]
  },
  "readOnlyRootFilesystem": true,
  "runAsNonRoot": true,
  "runAsUser": 101
}

```

```
Summary
========================

Target:     minikube
Score:      C    55/100     (100% completed)
✓ Passed:   ██████ 39% (41)
✕ Failed:   ████ 27% (29)
! Errors:   █████ 33% (35)
» Skipped:  1% (1)

Policies:
F   0  Platform Vulnerability Policy by Mondoo
.  ..  Platform Overview Information by Mondoo
B  65  Linux Security Baseline by Mondoo
A 100  Platform End-of-Life Policy by Mondoo

```

The scans ran on my local minikube. The rootless image was `mondoo/client:6.6.0-rootless`.

**Comment by imilchev** [comment]
I just verified that there is actually a difference. It depends on what policies you have enabled in your space and what type of checks they do. We can check a lot without being `root` but there are most certainly things that require `root` access.

```
Summary
========================

Target:     minikube
Score:      B    60/100     (100% completed)
✓ Passed:   ███████ 50% (92)
✕ Failed:   ████ 25% (46)
! Errors:   ███ 22% (41)
» Skipped:  3% (6)

Policies:
.  ..  Platform Overview Information by Mondoo
F   0  Platform Vulnerability Policy by Mondoo
C  52  CIS Kubernetes V1.23 Benchmark - Level 2 - Worker Node
C  55  CIS Kubernetes V1.23 Benchmark - Level 1 - Worker Node
B  65  Linux Security Baseline by Mondoo
B  75  CIS Kubernetes V1.23 Benchmark - Level 1 - Master Node
B  76  CIS Kubernetes V1.23 Benchmark - Level 2 - Master Node
A 100  Platform End-of-Life Policy by Mondoo

Report URL: https://console.mondoo.com/space/fleet/2BfB3Xlv1IKib3F4gVVUVcCECvh?spaceId=musing-boyd-615534
/ # whoami
root
```

```
Summary
========================

Target:     minikube
Score:      C    59/100     (100% completed)
✓ Passed:   ███████ 49% (90)
✕ Failed:   ████ 25% (46)
! Errors:   ███ 23% (43)
» Skipped:  3% (6)

Policies:
F   0  Platform Vulnerability Policy by Mondoo
.  ..  Platform Overview Information by Mondoo
C  47  CIS Kubernetes V1.23 Benchmark - Level 2 - Worker Node
C  50  CIS Kubernetes V1.23 Benchmark - Level 1 - Worker Node
B  65  Linux Security Baseline by Mondoo
B  75  CIS Kubernetes V1.23 Benchmark - Level 1 - Master Node
B  76  CIS Kubernetes V1.23 Benchmark - Level 2 - Master Node
A 100  Platform End-of-Life Policy by Mondoo

Report URL: https://console.mondoo.com/space/fleet/2BfB3Xlv1IKib3F4gVVUVcCECvh?spaceId=musing-boyd-615534
~ $ whoami
mondoo
```

**Comment by czunker** [comment]
Gave it another try with the CIS policies enabled.

Even with `CAP_SYS_ADMIN` we cannot evaluate all CIS rules. We need root for the node scans.

**Comment by imilchev** [comment]
I think this can be closed now since we merged #447
